### ISSUE TITLE
Expose emptyDir "Memory" mode for optionally faster IO from PFS in workers

### ISSUE THREAD

**Issue by Southclaws** [pachyderm/pachyderm#2048]
When the worker container downloads data into the `emptyDir`, IO speed could be optionally (via a pipeline flag) optimised for certain types of work that require frequent IO or caching.

Kubernetes exposes a [tmpfs](https://kubernetes.io/docs/concepts/storage/volumes/#emptydir) directory type for the `emptyDir` volume via `emptyDir.medium: Memory`

This would obviously increase RAM usage in the cluster but it's definitely worth adding a flag for as the implementation would be very minimal. I could see some use-cases benefiting from the tradeoff such as workflows that require lots of disk IO for large operations or cache usage.

In terms of raw benchmarks, I don't have any on hand but as far as I know the feature in Kubernetes is just an exposure of a Linux feature so there are likely many resources and comparisons for the benefits and drawbacks. It will also differ greatly by cloud service and machine type.

**Comment by sjezewski** [comment]
Interesting suggestion. We haven't seen a lot of interest in this, so I'm marking it as stale. But if we see more interest, this could be something we could support

**Comment by claytonlemons** [comment]
@sjezewski There's huge interest in this feature from my team. We run our pipelines on larger nodes where many pipeline workers contend for the same underlying disk that backs the `pachyderm_worker` volume, which is mounted to `/pfs`

We tried a pipeline both with a memory dir `pod_patch` and without.
- The pipeline processed 1000, 4MB files, each of which generate ~68K output files
- The pipeline ran with 126 workers, which filled up 2, 64-core/256 GB nodes backed by gp2 EBS volumes.
- The `pod_patch` below creates a memory-backed volume for `/pfs`
   ```pod_patch: >-
   [
     {
       "op": "add",
       "path": "/volumes/1/emptyDir/medium",
       "value": "Memory"
     }
   ]
  ```

The results are astonishing:
- The pipeline as described above ran in 10 minutes _with_ the pod_patch
- Without the `pod_patch` the same pipeline ran for 12 hours, processing most of the input data, but eventually failed on a datum due to IO issues (likely because the underlying EBS volume couldn't handle the load).

Long story short, memory-backed volumes == up to a 72x improvement (and possibly more)!
